inference-engine/src/gna_plugin/frontend/quantized_layer_params.hpp

```diff
@@ -7,9 +7,32 @@
 namespace GNAPluginNS {
 
 struct Quantization {
+public:
+    float GetScale() const {
+        return scale;
+    }
+    void SetScale(float s) {
+        scale = s;
+        scale_set = true;
+    }
+    bool IsScaleSet() const {
+        return scale_set;
+    }
+    float GetMaxValue() const {
+        return max_value;
+    }
+    void SetMaxValue(float value) {
+        max_value = value;
+        max_value_set = true;
+    }
+    bool IsMaxValueSet() const {
+        return max_value_set;
+    }
+private:
     float scale = 1.0f;
-    float offset = 0.0f;
-    int shift = 0.0f;
+    bool scale_set = false;
```

Comment:
what if scale not set?  - simply do not inject this structure or use scale 0.0
How is that different from scale 1.0? If you try to minimize number of quantisation calls i dont think this needed - this just a simple container for scales - max value is sort of statistics please don't mix.  
I gues you also use this as a walker helper to avoid loops etc - this also looks a helper and dont need to be in public interface for quantparams.
I prefer Inject other structure like quantstat if you need, also quant stat can be implemented using unordered_map with layer's names - we will further switch to FQ layers anyway and this injections might not needed in future.